Objects/unicodeobject.c

```diff
@@ -768,7 +768,7 @@ ensure_unicode(PyObject *obj)
 {
     if (!PyUnicode_Check(obj)) {
         PyErr_Format(PyExc_TypeError,
-                     "must be str, not %T", obj);
+                     "must be str, not %t", obj);
```

Comment:
What was used instead of %t/%T before?

Reply:
Python 3.7 code:

```
        PyErr_Format(PyExc_TypeError,
                     "must be str, not %.100s",
                     Py_TYPE(obj)->tp_name);
```

Reply:
Thus it was closer to %T. In error messages it is better to use fully qualified names.